File: apps/frontend/lib/src/features/auth/data/dto/login_dto.dart
class LoginDto {
  LoginDto({
    required this.siteCode,
    required this.email,
    required this.password,
    this.companyId,
  });

  final String siteCode;
  final String email;
  final String password;
  final String? companyId;

  factory LoginDto.fromJson(Map<String, dynamic> json) => LoginDto(
        siteCode: json['siteCode'] as String,
        email: json['email'] as String,
        password: json['password'] as String,
        companyId: json['companyId'] as String?,
      );

  Map<String, dynamic> toJson() => {
        'siteCode': siteCode,
        'email': email,
        'password': password,
        'companyId': companyId,
      };
}

File: apps/backend/src/modules/auth/dto/login.dto.ts
import { IsEmail, IsString, MinLength, IsOptional } from 'class-validator';
import { ApiProperty } from '@nestjs/swagger';

export class LoginDto {
  @ApiProperty({ example: 'WIPRP-HQ', description: 'Site code for tenant scoping' })
  @IsString()
  siteCode!: string;

  @ApiProperty({ example: 'user@example.com' })
  @IsEmail()
  email!: string;

  @ApiProperty({ example: 'password123' })
  @IsString()
  @MinLength(8)
  password!: string;

  @ApiProperty({ example: '123e4567-e89b-12d3-a456-426614174000', required: false })
  @IsString()
  @IsOptional()
  companyId?: string;
}

export class RefreshTokenDto {
  @ApiProperty()
  @IsString()
  refreshToken!: string;
}

export class OAuth2CallbackDto {
  @ApiProperty()
  @IsString()
  code!: string;

  @ApiProperty()
  @IsString()
  @IsOptional()
  state?: string;
}

